### PR TITLE
BC-3479 - a11y interactive elements

### PR DESCRIPTION
**Pull request by odalys-dataport**
# Short Description
Improve accessibility for interactive elements 

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-3479

## Changes
Besides the a11y changes this PR also
- adapts injection keys to follow our naming convention "xyz_KEY"
- moves components out of the legacy folder to a more appropriate destination
- rewrites some components to composition API and TS
- removes unused translation keys

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/store/courses.js
Comment (christian-darsow) [inline src/store/courses.js:1]: Just old stuff that is never needed again?

File: src/utils/inject/injection-keys.ts
Comment (christian-darsow) [inline src/utils/inject/injection-keys.ts:1]: I would suggest to at least mention all these injection related changes in the description of the PR

File: src/store/courses.js
Comment (odalys-dataport) [reply src/store/courses.js:1]: Yeah, I couldn't see any usage of this store and anything that will happen in this context in the future will happen in rooms and boards.

File: src/components/topbar/PopupIconInitials.vue [hunk 1]
```diff
@@ -5,10 +5,16 @@
...
-			:title="`${firstName} ${lastName} (${userRole})`"
-			:aria-label="`${$t(
-				'global.topbar.initials.currentUser'
-			)} ${firstName} ${lastName} ${userRole}`"
+			:title="
+				$t('global.topbar.userMenu.ariaLabel', {
+					userName: `${firstName} ${lastName} (${userRole})`,
+				})
+			"
+			:aria-label="
```
Comment (christian-darsow) [inline src/components/topbar/PopupIconInitials.vue:13]: Its best to have both? title and aria-label with the same value in this case. Also just to understand better :)

File: src/components/topbar/TheTopBar.vue
Comment (christian-darsow) [inline src/components/topbar/TheTopBar.vue:1]: As with the injection changes. can we state in the description what we move out of the legacy? or why other stuff stays in "legacy" like TheSidebar in contrast to this component

File: src/components/molecules/RoomDotMenu.vue [hunk 2]
```diff
@@ -10,11 +10,11 @@
 	>
 		<template #activator="{ on, attrs }">
 			<v-btn
-				v-show="show"
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/molecules/RoomDotMenu.vue:13]: After also checking it was used before but always set to `true`

File: src/components/molecules/ContentSearchbar.vue [hunk 3]
```diff
@@ -4,7 +4,7 @@
 			<input
 				ref="searchInput"
 				:value="value"
-				:aria-label="ariaLabel"
+				:aria-label="$t('pages.content.index.search.placeholder')"
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/molecules/ContentSearchbar.vue:7]: Removed the ariaLabel property as never used in the component

File: src/components/legacy/TheSidebar.vue [hunk 4]
```diff
@@ -33,6 +33,7 @@
 							:to="route.to"
 							:href="route.href"
 							:no-styles="true"
+							:aria-label="$t(route.title)"
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/legacy/TheSidebar.vue:36]: On smaller screens the visible part of the link is unvisible and only an icon is shown. Therefore an aria-label is necessary

File: src/utils/inject/injection-keys.ts
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/utils/inject/injection-keys.ts:1]: Added some info in the changes section

File: src/components/topbar/TheTopBar.vue
Comment (christian-darsow) [reply src/components/topbar/TheTopBar.vue:1]: Added some info in the changes section

File: src/components/topbar/TheTopBar.vue [hunk 5]
```diff
@@ -54,23 +54,22 @@
...
-					icon="help_area_outline"
+					:icon="mdiHelpCircleOutline"
 					:title="$t('global.topbar.actions.helpSection')"
 					:aria-label="$t('global.topbar.actions.helpSection')"
 					data-test-id="help-btn"
 				>
 					<help-dropdown />
 				</popup-icon>
-				<div v-if="school" class="school-name item">
+				<div v-if="school" class="school-name item" tabindex="0">
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/topbar/TheTopBar.vue:64]: We could not figure out, why this tabbable text could help for a11y and decided to remove for now

File: src/components/organisms/ContentCard.vue [hunk 6]
```diff
@@ -28,7 +28,7 @@
 							<img
 								:src="thumbnail()"
 								class="content__img-thumbnail"
-								:alt="$t('pages.content.card.img.alt')"
+								alt=""
```
[the remark this answers is not in the document]
Comment (christian-darsow) [reply src/components/organisms/ContentCard.vue:31]: Lets take that to the a11y experts and ask if an empty alt is the way to go when not having a alt text, as here not provided by external sources

File: src/components/topbar/PopupIconInitials.vue (hunk 1 above)
Comment (christian-darsow) [reply src/components/topbar/PopupIconInitials.vue:13]: Following this https://www.aditus.io/aria/aria-label/#aria-label-vs-title we decided to remove the title and keep the new aria-label

**Review by christian-darsow — DISMISSED** [review]

File: src/store/courses.js
Comment (christian-darsow) [reply src/store/courses.js:1]: The courses store is used in the `AddContentModal.vue` and was re-introduced

**Review by MartinSchuhmacher — APPROVED** [review]

**Review by christian-darsow — APPROVED** [review]

**Pull request merged**